File: cpp/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        vector<int> results;
        
        for(int i = 0; i < nums.size(); ++i){
            int comp = target - nums[i];
            if (umap.find(comp) != umap.end()){
                results.push_back(umap[comp]);
                results.push_back(i);
                return results;
            }
            umap[nums[i]] = i; 
        }
        return results;
    }
};

File: java/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        int results[] = new int[2];
        
        for(int i = 0; i < nums.length; ++i){
            int comp = target - nums[i];
            if (hmap.containsKey(comp)){
                results[0] = hmap.get(comp);
                results[1] = i;
                return results;
            }
            hmap.put(nums[i], i);
        }
        return results;
    }
}